### ISSUE TITLE
Allow EditableAttribute for classes

### ISSUE THREAD

**Issue by cervengoc** [dotnet/corefx#10527]
I have a scenario where I need to specify `EditableAttribute` on a class, instead of applying it separately to every property. Is there any drawback of allowing this attribute for classes too?

**Comment by krwq** [comment]
Hello @cervengoc, this is simply work to be done. Also there is a lot of cases which needs to be considered and thought through, i.e: what happens when someone inherits from the class with this attribute; other attributes which are not currently applicable for classes - should they be applicable to classes too etc..

I'll mark this as Future for now. If you strongly feel like this should be here soon I recommend to start API design issue with proposed changes

**Comment by ajcvickers** [comment]
@cervengoc We have reviewed the EditableAttribute and it seems that it is geared towards properties and not classes. In particular the AllowInitialValue makes sense for a property but is, at best, ambiguous if applied to a class.

Also, it would take consumers of the annotation to read it from the class before it could be used for anything, and it seems unlikely that many (if any) consumers would do this even if we made the change.

For these reasons we are not planning to do this.
